mitmproxy/addons/next_layer.py

```diff
@@ -350,10 +350,15 @@ def _setup_reverse_proxy(context: Context, data_client: bytes) -> Layer:
                     stack /= ClientTLSLayer(context)
                 stack /= HttpLayer(context, HTTPMode.transparent)
             case "https":
-                stack /= ServerTLSLayer(context)
-                if starts_like_tls_record(data_client):
-                    stack /= ClientTLSLayer(context)
-                stack /= HttpLayer(context, HTTPMode.transparent)
+                if _starts_like_quic(data_client):
```

Comment:
Can we inspect client_conn.transport_protocol instead? That feels way more robust.